dnf-behave-tests/features/stick-to-vendor.feature

```diff
@@ -0,0 +1,67 @@
+Feature: DNF allow_vendor_change option in dnf.conf
+#
+# Vendor: First Vendor
+# Repo  : dnf-ci-vendor-1
+#   vendor-1.0
+# Repo  : dnf-ci-vendor-1-updates
+#   vendor-1.1
+#
+# Vendor: Second Vendor
+# Repo  : dnf-ci-vendor-2
+#   vendor-1.0
+# Repo  : dnf-ci-vendor-2-updates
+#   vendor-1-1
+#
+Background:
+  Given I use repository "dnf-ci-vendor-1"
+  #Given I use repository "dnf-ci-vendor-1-updates"
+  #Given I use repository "dnf-ci-vendor-2"
+  #Given I use repository "dnf-ci-vendor-2-updates"
+   And I create and substitute file "/etc/dnf/dnf.conf" with
+   """
+   [main]
+   allow_vendor_change=False
+   """
+   And I successfully execute dnf with args "install vendor"
+   Then the exit code is 0
+   And transaction is following
+       | Action  | Package             |
+       | install | vendor-1.0-1.x86_64 |
+
+  Given I successfully execute rpm with args "-qi vendor"
+   Then the exit code is 0
+   And stdout contains "Vendor      : First Vendor"
+
+@bz1788371
+Scenario: Upgrade sticks to vendor
+  Given I use repository "dnf-ci-vendor-1-updates"
+  Given I use repository "dnf-ci-vendor-2-updates"
+   When I execute dnf with args "upgrade vendor"
+   Then the exit code is 0
+   And transaction is following
+       | Action  | Package             |
+       | upgrade | vendor-1.1-1.x86_64 |
+  Given I successfully execute rpm with args "-qi vendor"
+   Then the exit code is 0
+   And stdout contains "Vendor      : First Vendor"
+
+@wip
+@bz1788371
+Scenario: No upgrade if same vendor not found
+  Given I use repository "dnf-ci-vendor-2-updates"
+   When I execute dnf with args "upgrade vendor"
+   Then the exit code is 0
+   And transaction is empty
+  Given I successfully execute rpm with args "-qi vendor"
+   Then the exit code is 0
+   And stdout contains "Vendor      : First Vendor"
+
+@bz1788371
+Scenario: No downgrade if same vendor not found
+  Given I use repository "dnf-ci-vendor-1-updates"
+  Given I use repository "dnf-ci-vendor-2"
+   When I execute dnf with args "install vendor-1.1-1"
+  Given I drop repository "dnf-ci-vendor-1"
+   When I execute dnf with args "downgrade vendor"
+   Then the exit code is 0
+   And transaction is empty
```

Comment:
This ends with:
```Error: 
 Problem: problem with installed package vendor-1.1-1.x86_64
  - cannot install both vendor-1.0-1.x86_64 and vendor-1.1-1.x86_64
  - conflicting requests
```
The error message indicates that it attempts to install both old and new package, possibly with a different vendor.
Please follow up with @j-mracek - he might know what's wrong. I suspect the wrong preselection of downgrade candidates.